Providers/nxOMSAutomationWorker/automationworker/worker/linuxutil.py

```diff
@@ -227,6 +227,28 @@ def get_lsb_release():
 
 
 @posix_only
+def get_oms_agent_id():
+    """Gets the oms agent id.
+
+    Returns:
+        string, the agent id
+        None,   if no agent id can be fouud
+    """
+    omsadmin_filepath = "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
```

Comment:
Our resource works with only primary workspace by design.  Only one instance of worker can run on a machine, we don't deal with selecting which secondary workspace to work with. Without a primary workspace, the resource does a no-op.